File: leetcode/power_of_four.erl
%% https://leetcode.com/problems/power-of-four/
-module(power_of_four).

-export([is_power_of_four/1]).

-spec is_power_of_four(N :: integer()) -> boolean().
is_power_of_four(N) when N == 4; N == 1 ->
    true;
is_power_of_four(N) when N < 4, N > -4 ->
    false;
is_power_of_four(N) ->
    case N rem 4 == 0 of 
        true -> is_power_of_four(N div 4);
        false -> false
    end.

File: leetcode/_Other/power_of_four.py
# https://leetcode.com/problems/power-of-four/

class Solution:
    def isPowerOfFour(self, n: int) -> bool:
        a = bin(n)[2:]
        if a[0] != '1' or len(a[1:]) % 2 != 0:
            return False
        for i in range(1, len(a)):
            if a[i] != '0':
                return False
        return True
